lib/numo/sparse/base.rb

```
@@ -37,6 +37,47 @@ def ndim
         return dtype if dtype < Numo::NArray
         raise ArgumentError, "Invalid dtype: #{dtype}"
       end
+
+      # Multiplies matrix by matrix
+      # @param matrix [sparse matrix] used to mulitiply original matrix
+      # @return [Sparse matrix] returns multiplied matrix
+      # @example
+      #   csr = Numo::Sparse::CSR.new(Numo::DFloat[[1, 0, 1], [1, 1, 1], [0, 0, 1]])
+      #   csc = Numo::Sparse::CSC.new(Numo::DFloat[[2, 2, 0], [0, 0, 2], [0, 0, 0]])
+      #   csr2 = csr.multiply(csc)
+      #   # => data => [2, 2, 2, 2, 2], indices => [0, 1, 0, 1, 2], indptr => [0, 2, 5, 5]
+      def multiply(matrix)
+        data1, indices1, indptr1, shape1= matrix.data, matrix.indices, matrix.indptr, matrix.shape
+        if shape[1] == shape1[0]
+          matrix1 = self.class.new(data, indices, indptr, shape).to_narray
+          matrix2 = matrix.class.new(data1, indices1, indptr1, shape1).to_narray
+          new_row, new_col, matrix_row, matrix_col, matrix1_row, matrix1_col, new_matrix_data = 0, 0, 0, 0, 0, 0, 0
+          shape2 = [shape[0], shape1[1]]
+          narray = data.class.zeros(shape2)
+          until new_row == shape[0] do
+            until new_col == shape1[1]do
+              until matrix_col == shape1[0] do
+                new_matrix_data += (matrix1[matrix_row, matrix_col] * matrix2[matrix1_row, matrix1_col])
+                matrix_col += 1
+                matrix1_row += 1
+              end
+              narray[new_row, new_col] = new_matrix_data
+              new_col += 1
+              matrix1_col += 1
+              new_matrix_data = 0
+              matrix_col = 0
+              matrix1_row = 0
+            end
+            new_col = 0
+            matrix1_col = 0
+            matrix_row += 1
+            new_row += 1
+          end
+          self.class.new(narray)
+        else
+        p "Unable to perform matrix multiplication"
```

Could you raise an error by the code below?

```ruby
raise Numo::NArray::ShapeError, "Unable to multiply matrix due to unmatched shapes"
```

---

And, please fix the indentation of this line.